File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer = arr;
    
    auto minPointer = min_element(answer.begin(), answer.end());
    
    int minValue = *minPointer;
    
    cout << minValue;
    
    answer.erase(minPointer);
    
    if(answer.empty()){
        answer.push_back(-1);
    }
    
    
    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {

        if(arr.length == 1){
            return new int[]{-1};
        }
        
        int[] answer = new int[arr.length - 1];
        int min = arr[0];
        for(int i=0; i<arr.length; i++){
            if(arr[i] < min){
                min  = arr[i];
            }
        }
        
        int index=0;
        for(int i=0; i<arr.length; i++){
            if(arr[i] != min){
                answer[index++] = arr[i];
            }
        }
        
        return answer;
        
    }
}